src/compiler/checker.ts

```diff
@@ -46005,7 +46005,13 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             languageVersion >= ScriptTarget.ES5 && name.escapedText === "Object"
             && host.getEmitModuleFormatOfFile(getSourceFileOfNode(name)) < ModuleKind.ES2015
         ) {
-            error(name, Diagnostics.Class_name_cannot_be_Object_when_targeting_ES5_with_module_0, ModuleKind[moduleKind]); // https://github.com/Microsoft/TypeScript/issues/17494
+            // https://github.com/Microsoft/TypeScript/issues/17494
+            error(
+                name,
+                Diagnostics.Class_name_cannot_be_Object_when_targeting_0_with_module_1,
+                ScriptTarget[languageVersion],

```

Comment:
We shouldn't be using reverse mapping to find the names of language versions, as this depends on how the enum is defined and will break if two things alias each other. We should have some other standard way to print out target kinds like this.

I'd personally suggest just making the message say "ES5 and above".